### PR TITLE
Refactor MTE-183 [v109] NavigationTest, ThirdPartySearchTest and PrivateBrowsingTest for M1

### PR DESCRIPTION
**Pull request by clarmso**
Let me try this again after pulling some new changes from `main`. 😬 I've picked the good changes from https://github.com/mozilla-mobile/firefox-ios/pull/12521

These tests are updated largely because of the M1 specific issues: Copy & Paste. There are some updates related to the updated user interface.

https://mozilla-hub.atlassian.net/browse/MTE-183
Full function test results to-date: https://storage.googleapis.com/mobile-reports/public/firefox-ios-M1/FullFunctionalTests/result_68/build/reports/tests.html

### CODE REVIEW

**Comment by clarmso** [comment]
Should I disable test that shows [FXIOS-5349](https://mozilla-hub.atlassian.net/browse/FXIOS-5349)? Alternatively, I can put the workaround.

File: Tests/XCUITests/PhotonActionSheetTest.swift [hunk 1]
```diff
@@ -34,35 +34,36 @@ class PhotonActionSheetTest: BaseTestCase {
...
     func testSendToDeviceFromPageOptionsMenu() {
         // User not logged in
         navigator.openURL(path(forTestPage: "test-mozilla-org.html"))
         waitUntilPageLoad()
-        waitForExistence(app.buttons[AccessibilityIdentifiers.Toolbar.shareButton], timeout: 10)
-        app.buttons[AccessibilityIdentifiers.Toolbar.shareButton].tap()
-        waitForExistence(app.buttons["Send Link to Device"], timeout: 10)
+        navigator.goto(BrowserTabMenu)
+        navigator.performAction(Action.ShareBrowserTabMenuOption)
+        waitForExistence(app.buttons["Send Link to Device"], timeout: TIMEOUT)
```
Comment (clarmso) [inline Tests/XCUITests/PhotonActionSheetTest.swift:66]: This test started to fail recently. The menu item "Send Link to Device" is no longer in the browser tab menu, but we can first click "Share" and then "Send Link to Device".

File: Tests/XCUITests/PhotonActionSheetTest.swift [hunk 2]
```diff
@@ -34,35 +34,36 @@ class PhotonActionSheetTest: BaseTestCase {
...
+            waitForExistence(app.buttons["Close"], timeout: TIMEOUT)
         }
     }
 
     func testSendToDeviceFromPageOptionsMenu() {
         // User not logged in
         navigator.openURL(path(forTestPage: "test-mozilla-org.html"))
         waitUntilPageLoad()
-        waitForExistence(app.buttons[AccessibilityIdentifiers.Toolbar.shareButton], timeout: 10)
-        app.buttons[AccessibilityIdentifiers.Toolbar.shareButton].tap()
```
Comment (clarmso) [inline Tests/XCUITests/PhotonActionSheetTest.swift:64]: Ah...some changes from upstream coming in around the same time when this branch is created.

File: Tests/XCUITests/PhotonActionSheetTest.swift [hunk 3]
```diff
@@ -34,35 +34,36 @@ class PhotonActionSheetTest: BaseTestCase {
...
         // Wait to see the Share options sheet
-        waitForExistence(app.buttons["Copy"], timeout: 10)
+        waitForExistence(app.buttons["Copy"], timeout: TIMEOUT)
     }
 
     // Smoketest
     func testShareOptionIsShownFromShortCut() {
         navigator.openURL(path(forTestPage: "test-mozilla-org.html"))
         waitUntilPageLoad()
-        waitForExistence(app.buttons[AccessibilityIdentifiers.Toolbar.shareButton], timeout: 10)
```
Comment (clarmso) [inline Tests/XCUITests/PhotonActionSheetTest.swift:48]: XCode keeps complaining about the `shareButton`.  It looks like this is introduced via a change after this branch was created.

**Comment by clarmso** [comment]
I worked on PhotonActionSheetTest but my work has be superseded.

**Comment by isabelrios** [comment]
> Should I disable test that shows [FXIOS-5349](https://mozilla-hub.atlassian.net/browse/FXIOS-5349)? Alternatively, I can put the workaround.

I would say to keep it as it is... once the issue is fixed it should pass..if not we will know that something is still wrong...

**Review by isabelrios — APPROVED** [review]
Let's land this changes to fix a few more tests.
There is one test faling in the PrivateBrowsing test suite: `testClosePrivateTabsOptionClosesPrivateTabs`
I think we need to update its behaviour as it fails when checking the number of tabs open. If I remember correctly there is an issue for that, I will try to find it.

**Pull request merged**